java/code/src/com/suse/manager/webui/websocket/VirtNotifications.java

```diff
@@ -74,6 +75,21 @@
     private static Map<Session, Set<Long>> wsSessions = new HashMap<>();
     private static Set<Session> brokenSessions = new HashSet<>();
 
+    // Map each virtualization action types to an id supplier. The id is indeed
+    // different for guests, pools, volumes and networks
+    private static final Map<Class<? extends Action>, Function<Action, String>> IDS_MAPPER = Map.of(
+            BaseVirtualizationAction.class,
+            (action) -> {
+                String id = "new-" + action.getId();
+                BaseVirtualizationAction virtAction = (BaseVirtualizationAction)action;
+                String uuid = virtAction.getUuid();
+                if (uuid != null) {
+                    id = uuid;
+                }
+                return id;
+            }
+        );
```

Comment:
I didn't fully understand why this is implemented as a `Map` (considering it only has one entry).

IIUC, `BaseVirtualizationAction` is meant to be the base class for all the virtualization actions. Maybe this logic can be implemented as a method in `BaseVirtualizationAction` and overriden in subclasses as needed.

Reply:
it's a map because I'll add another entry for the pool notifications in another PR ;)

Reply:
See [this commit](https://github.com/uyuni-project/uyuni/pull/1166/commits/dfbb4672c37a5e65a866da87f6880a7f3ecd65c1) for another entry in this map.

The problem is that pool actions have a different base action class. `BaseVirtualizationAction` is the legacy class, but is badly named since it should rather be `BaseVirtualizationGuestAction`

Reply:
Ok, makes sense now. Thanks.